src/gauss_adjoint.jl

```diff
@@ -436,9 +793,77 @@ end
 function vec_pjac!(out, λ, y, t, S::GaussIntegrand)
     @unpack pJ, pf, p, f_cache, dgdp_cache, paramjac_config, sensealg, sol = S
     f = sol.prob.f
+    
+    isautojacvec = get_jacvec(sensealg)
+    # y is aliased
+
+    if !isautojacvec
+        if DiffEqBase.has_paramjac(f)
+            f.paramjac(pJ, y, p, t) # Calculate the parameter Jacobian into pJ
+        else
+            pf.t = t
+            jacobian!(pJ, pf, p, f_cache, sensealg, paramjac_config)
+        end
+        mul!(out', λ', pJ)
+    elseif sensealg.autojacvec isa ReverseDiffVJP
+        tape = paramjac_config
+        tu, tp, tt = ReverseDiff.input_hook(tape)
+        output = ReverseDiff.output_hook(tape)
+        ReverseDiff.unseed!(tu) # clear any "leftover" derivatives from previous calls
+        ReverseDiff.unseed!(tp)
+        ReverseDiff.unseed!(tt)
+        ReverseDiff.value!(tu, y)
+        ReverseDiff.value!(tp, p)
+        ReverseDiff.value!(tt, [t])
+        ReverseDiff.forward_pass!(tape)
+        ReverseDiff.increment_deriv!(output, λ)
+        ReverseDiff.reverse_pass!(tape)
+        copyto!(vec(out), ReverseDiff.deriv(tp))
+    elseif sensealg.autojacvec isa ZygoteVJP
+        _dy, back = Zygote.pullback(p) do p
+            vec(f(y, p, t))
+        end
+        tmp = back(λ)
+        #out[:] .= vec(tmp[1])
+        recursive_copyto!(out,tmp[1])
+    elseif sensealg.autojacvec isa EnzymeVJP
+        tmp3, tmp4 = paramjac_config
+        tmp4 .= λ
+        out .= 0
+        Enzyme.autodiff(Enzyme.Reverse, pf, Enzyme.Duplicated(tmp3, tmp4),
+            y, Enzyme.Duplicated(p, out), t)
+    else
+        error("autojacvec choice $(sensealg.autojacvec) is not supported by GaussAdjoint")
+    end
+    # TODO: Add tracker?
+    
+    
+    return out
+end
+
+function vec_pjac_diffusion!(out, λ, y, t, S::GaussIntegrand, W = nothing)
+    
+    @unpack pJ, pf, p, f_cache, dgdp_cache, paramjac_config, sensealg, sol = S
+    f = sol.prob.f
+    g = sol.prob.g
     isautojacvec = get_jacvec(sensealg)
     # y is aliased
 
+    if sensealg.autojacvec isa ZygoteVJP
+        if W === nothing
+            _dy, back = Zygote.pullback(y, p) do u, p
+                vec(g(u, p, t))
```

Comment:
I think you want to compute the following:
`vec(g(u, p, t)*dW)`

(sorry for the slow responses. I'm catching up rn -- got a bad cold.)

Reply:
I tried this too (see most recent commit) but don't seem to get the right results

Reply:
Did you check if it's the correct dW that's extracted?

Reply:
I guess I am a but confused because $dW$ should be a function of $t$, right? The SDE solution type only stores $W$ as a function of time. Should I use $dW_i = W_{t_{i+1}} - W_{t_i}$, or am I misunderstanding something?

Reply:
Yes, in your current version, you assume `dW` to be constant. If you use several Gauss points for the integral (how many do you use?), you should probably compute dW appropriately for those times.

Reply:
For SDEs, it defaults to 10 since the SDE integrator algorithms do not have an order. 

What do you mean compute $dW$ appropriately?

Reply:
they do have an order -- EM is 0.5 strong/ 1 weak 😅 


 $dW_i = W_{t_{i+1}} - W_{t_i}$

Reply:
Ok thanks! Should these be the $t_{i}$'s from the forward pass? Then in the backwards pass, should the $t$'s that the solver stops at be identical to the forward pass?

Reply:
I think for now the best thing to do to check the implementation is to make sure it only uses $t$ values that were encountered in the forward pass -- in general I think it's not strictly necessary because we can draw from a Brownian bridge... but this likely increases the variance of the estimate.

Reply:
that makes sense, thanks. I will try to implement this now

Reply:
I can force the solver stop at the $t$ values from the forward pass but I don't think it is possible to ensure that the Gauss points for computing the integral are always at these points. Should I just interpolate or will I have to use a Brownian bridge?